addons/sale_mrp/models/sale_mrp.py

```diff
@@ -10,20 +10,35 @@ class SaleOrderLine(models.Model):
 
     @api.multi
     def _compute_qty_delivered(self):
-        super(SaleOrderLine, self)._compute_qty_delivered()
-
-        for line in self:
-            if line.qty_delivered_method == 'stock_move':
-                # In the case of a kit, we need to check if all components are shipped. Since the BOM might
-                # have changed, we don't compute the quantities but verify the move state.
-                bom = self.env['mrp.bom']._bom_find(product=line.product_id)
+        order_line_ids = self.env['sale.order.line']
+        for order_line in self:
+            if order_line.qty_delivered_method == 'stock_move':
+                bom = self.env['mrp.bom']._bom_find(product=order_line.product_id)
                 if bom and bom.type == 'phantom':
-                    moves = line.move_ids.filtered(lambda m: m.picking_id and m.picking_id.state != 'cancel')
-                    bom_delivered = all([move.state == 'done' for move in moves])
-                    if bom_delivered:
-                        line.qty_delivered = line.product_uom_qty
+                    order_line_ids |= order_line
+                    moves = order_line.move_ids.filtered(lambda m: m.picking_id and m.picking_id.state != 'cancel' and not m.scrapped)
+                    qty_ratios = []
+                    boms, bom_sub_lines = bom.explode(order_line.product_id, order_line.product_uom_qty)
+                    for bs_line in bom_sub_lines:
+                        bom_line = bs_line[0]
+                        bom_line_datas = bs_line[1]
+                        if bom_line.product_id.id in moves.mapped('product_id').mapped('id'):
```

Comment:
bom_line.product_id in moves.mapped('product_id')

or if you really want but useless: bom_line.product_id.id in moves.mapped('product_id.id')